cmd/pdptool/main.go

```diff
@@ -1140,20 +1161,37 @@ var addRootsCmd = &cli.Command{
 			Usage:    "Root CID and its subroots. Format: rootCID:subrootCID1+subrootCID2,...",
 			Required: true,
 		},
+		&cli.StringFlag{
+			Name:     "extra-data",
+			Usage:    "Optional ABI-encoded extra data as a hex string to pass to the listener contract (max 2048 bytes)",
+			Required: false,
+		},
 	},
 	Action: func(cctx *cli.Context) error {
 		serviceURL := cctx.String("service-url")
 		serviceName := cctx.String("service-name")
 		proofSetID := cctx.Uint64("proof-set-id")
 		rootInputs := cctx.StringSlice("root")
+		extraDataHexStr := cctx.String("extra-data")
 
-		// Load the private key (implement `loadPrivateKey` according to your setup)
+		// Validate extraData hex string and its decoded length
+		if extraDataHexStr != "" {
```

Comment:
nit: extract to a small validation and decode function since we use this code a bunch